src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3094,12 +3094,21 @@ private void AddSynthesizedRecordMembersIfNecessary(MembersAndInitializersBuilde
 
             return;
 
-            SynthesizedRecordConstructor addCtor(RecordDeclarationSyntax declWithParameters)
+            SynthesizedRecordConstructor? addCtor(RecordDeclarationSyntax declWithParameters)
             {
                 Debug.Assert(declWithParameters.ParameterList is object);
                 var ctor = new SynthesizedRecordConstructor(this, declWithParameters, diagnostics);
-                members.Add(ctor);
-                return ctor;
+
+                if (ctor.ParameterCount == 1 && ctor.Parameters[0].Type.Equals(this))
```

Comment:
>Equals(this) [](start = 72, length = 12)

All aspects should be ignored here, I think.

Reply:
We're already ignoring differences that can appear in source code (nullability). But I think you're right in terms of future-proofing. Thanks

---
In reply to: [543392514](https://github.com/dotnet/roslyn/pull/49975#discussion_r543392514) [](ancestors = 543392514)